### PR TITLE
Fix and test user purge for dashboard.email_preference

### PR DESCRIPTION
**Pull request by islemaster**
Fix and test user purge for dashboard.email_preference

From [spec](https://docs.google.com/spreadsheets/d/1RybuZunt5F6zSXp33oC4KdaBl20VMHVi0yprI8P30y0/edit#gid=670705846) (requires login).

Expected behavior: All EmailPreference rows associated with a purged user's email address are also deleted.

### CODE REVIEW

File: lib/cdo/delete_accounts_helper.rb [hunk 1]
```diff
@@ -186,6 +186,12 @@ def remove_census_submissions(email)
     Census::CensusSubmission.where(submitter_email_address: email).each(&:destroy)
   end
 
+  # Removes EmailPreference records associated with this email address.
+  # @param [String] email An email address
+  def remove_email_preferences(email)
+    EmailPreference.where(email: email).each(&:destroy)
```
Comment (sureshc) [inline lib/cdo/delete_accounts_helper.rb:192]: EmailPreference has a unique constraint on `email` so maybe use `find_by(email: email)`?

Comment (sureshc) [reply lib/cdo/delete_accounts_helper.rb:192]: I assume the email being passed as an argument by the caller came from the User model and has already been stripped and lower-cased?

Comment (islemaster) [reply lib/cdo/delete_accounts_helper.rb:192]: Hmm, good question.  I'll have to investigate.  I know we strip and downcase before we hash emails, but we might store the original formatting for display back to the user.

Comment (sureshc) [reply lib/cdo/delete_accounts_helper.rb:192]: Looks like User model strips and downcases before save:
https://github.com/code-dot-org/code-dot-org/blob/12f8a844cd81d85c00512c9c5e8c1f5f89c6dc21/dashboard/app/models/user.rb#L544-L562

**Pull request merged**